shpc/main/modules/base.py

```diff
@@ -106,24 +106,26 @@ def uninstall(self, name, force=False):
         self.container.delete(name)
 
         if container_dir != module_dir:
-            self._uninstall(container_dir, "$container_base/%s" % name, force)
+            force = force or self._uninstall(container_dir, "$container_base/%s" % name, force)
             self._uninstall(module_dir, "$module_base/%s" % name, force)
         else:
             self._uninstall(module_dir, "$module_base/%s" % name, force)
 
     def _uninstall(self, module_dir, name, force=False):
         """
         Sub function, so we can pass more than one folder from uninstall
+        Returns a boolean: whether the directory has been deleted or not
         """
         if os.path.exists(module_dir):
             if not force:
                 msg = "%s, and all content below it? " % name
                 if not utils.confirm_uninstall(msg, force):
-                    return
+                    return False
             self._cleanup(module_dir)
             logger.info("%s and all subdirectories have been removed." % name)
         else:
```

Comment:
I'm weary about this case - if the module directory doesn't exist we sort of "force" the uninstall to be True for the container directory no matter what, and I'm not sure that is best. E.g., I can see someone wanting to re-generate the module file, and they decide to delete the module first (but not the container) and choose yes for the first but not the second. 

If the person chooses `--force` then this should allow them to have a non-interactive "yes to all the things" install, otherwise I think the current setup is what I'd want.

Reply:
I'm not following you. shpc first deletes the container, and then the module, cf https://github.com/singularityhub/singularity-hpc/blob/main/shpc/main/modules/base.py#L109-L110

By the time shpc checks whether the module directory exists or not, it will have already processed the container directory

Reply:
Yep, so I think this design was done before we had two things to uninstall - e.g., the confirm was placed in uninstall because it was called first, and when I refactored to have an _uninstall called on the two directories I needed to move the confirmation and check of force up one level! So what I think we should do to fix this is to remove the confirmation from _uninstall and put into uninstall so _uninstall is called already having the confirmation, and from the main uninstall() we do both calls of _uninstall() (if relevant) in one swoop after that.